idrstream/CP_idr.py

```diff
@@ -0,0 +1,391 @@
+        # make directory for the input of cellprofiler and copy metadata file into images folder
+        metadata_save_path = pathlib.Path(
+            f"{self.CP_project_path}/inputs/images/{metadata_path.name}"
+        )
+        metadata_save_path.parents[0].mkdir(parents=True, exist_ok=True)
+        shutil.copyfile(metadata_path, metadata_save_path)
+
+        # make directory for the output of cellprofiler
+        output_save_path = pathlib.Path(f"{self.CP_project_path}/outputs/features/")
+        output_save_path.mkdir(parents=True, exist_ok=True)
+        self.CP_output_path = output_save_path
+
+        self.logger.info("Copied metadata file for CellProfiler run to use")       
+
+    def init_downloader(
+        self,
+        aspera_path: pathlib.Path,
+        aspera_key_path: pathlib.Path,
+        screens_path: pathlib.Path,
+    ):
+        """
+        initialize aspera downloader
+
+        Parameters
+        ----------
+        aspera_path : pathlib.Path
+            path to aspera sdk
+        aspera_key_path : pathlib.Path
+            path to aspera ssh key
+        screens_path : pathlib.Path
+            path to screens file used to locate IDR data
+        """
+        self.downloader = download.AsperaDownloader(
+            aspera_path, aspera_key_path, screens_path, self.idr_id
+        )
+        self.logger.info("Aspera downloader initialized")
+
+    def init_preprocessor(self, fiji_path: pathlib.Path):
+        """
+        initialize basicpy preprocessor
+
+        Parameters
+        ----------
+        fiji_path : pathlib.Path
+            path to Fiji.app folder
+        """
+        self.preprocessor = preprocess.BasicpyPreprocessor(fiji_path)
+        self.logger.info("Basicpy preprocessor initialized")
+
+    def convert_tsv_to_csv(
+        self, data_to_process_tsv: pathlib.Path, metadata_save_path: pathlib.Path
+    ):
+        """
+        Convert `data_to_process.tsv` file into a .csv for CellProfiler to read in when processing the images
+
+        Parameters
+        ----------
+        data_to_process_tsv : str
+            string path to the `data_to_process.tsv` file
+        """
+        # read in metadata tsv file
+        data_to_process_csv = pd.read_csv(data_to_process_tsv, sep="\t")
+
+        # converting tsv file into csv
+        data_to_process_csv.to_csv(metadata_save_path, index=False)
+
+        self.logger.info("CellProfiler metadata has been converted and saved")
+
+    def prepare_batch(self, batch_metadata: pd.DataFrame):
+        """
+        download data from a batch image and run further processes
+        image data is saved in the CP project folder
+
+        Parameters
+        ----------
+        batch_metadata : pd.DataFrame
+            metadata of images to extract features from
+        """
+        # iterate through image metadatas and download, preprocess, and segment each frame into the batch project
+        for index, row in batch_metadata.iterrows():
```

Comment:
It looked like `index` might not get used below. If it's not used, consider naming it `_` to indicate the throwaway nature of the index yield from `pd.DataFrame.iterrows`.

```suggestion
        for _, row in batch_metadata.iterrows():
```

Reply:
:+1: